src/Microsoft.AspNetCore.Blazor/Components/BlazorComponent.cs

```diff
@@ -158,18 +158,19 @@ public virtual void SetParameters(ParameterCollection parameters)
                 _hasCalledInit = true;
                 OnInit();
 
-                // If you override OnInitAsync and return a nonnull task, then by default
-                // we automatically re-render once that task completes.
+                // If you override OnInitAsync, then by default we automatically re-render once that task completes.
                 var initTask = OnInitAsync();
-                if (initTask != null && initTask.Status != TaskStatus.RanToCompletion)
+                if (initTask.Status != TaskStatus.RanToCompletion)
```

Comment:
Might be worthwhile to leave the null check in, in the event derived types followed this code and returned `null`